arcgis-ios-sdk-samples/Maps/Display layer view state/LayerStatusViewController.swift

```diff
@@ -15,73 +15,49 @@
 import UIKit
 import ArcGIS
 
-class LayerStatusViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
-    @IBOutlet private var mapView: AGSMapView!
-    @IBOutlet private var tableView: UITableView!
-    @IBOutlet private var visualEffectView: UIVisualEffectView!
+class LayerStatusViewController: UIViewController {
+    // MARK: Instance properties
     
-    private var map: AGSMap!
-    
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        
-        //constraint visual effect view to the map view's attribution label
-        let constraint = self.visualEffectView.bottomAnchor.constraint(equalTo: self.mapView.attributionTopAnchor, constant: -10)
-        
-        //activate constraint
-        constraint.isActive = true
-    }
-    
-    override func viewDidAppear(_ animated: Bool) {
-        super.viewDidAppear(animated)
-        
-        self.map = AGSMap()
-        
-        //create tiled layer using a url
-        let tiledLayer = AGSArcGISTiledLayer(url: URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/WorldTimeZones/MapServer")!)
-        //add the layer to the map
-        self.map.operationalLayers.add(tiledLayer)
-
-        //create an map image layer using a url
-        let imageLayer = AGSArcGISMapImageLayer(url: URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/Census/MapServer")!)
-        imageLayer.minScale = 40000000
-        imageLayer.maxScale = 2000000
-        //add it to the map
-        self.map.operationalLayers.add(imageLayer)
-        
-        //create feature layer using a url
-        let featureTable = AGSServiceFeatureTable(url: URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/Recreation/FeatureServer/0")!)
-        let featurelayer = AGSFeatureLayer(featureTable: featureTable)
-        //add it to the map
-        self.map.operationalLayers.add(featurelayer)
-        
-        //reload table
-        self.tableView.reloadData()
-        //assign map to the map view
-        self.mapView.map = self.map
-        //zoom to custom viewpoint
-        self.mapView.setViewpoint(AGSViewpoint(center: AGSPoint(x: -11e6, y: 45e5, spatialReference: .webMercator()), scale: 5e7))
-        
-        //layer status logic
-        //assign a closure for layerViewStateChangedHandler, in order to receive layer view status changes
-        self.mapView.layerViewStateChangedHandler = { [weak self] (layer: AGSLayer, state: AGSLayerViewState) in
-            DispatchQueue.main.async {
+    /// The map view managed by the view controller.
+    @IBOutlet weak var mapView: AGSMapView! {
+        didSet {
+            mapView.map = AGSMap(basemap: .topographic())
+            mapView.map?.operationalLayers.add(featureLayer)
+            mapView.map?.initialViewpoint = AGSViewpoint(center: AGSPoint(x: -11e6, y: 45e5, spatialReference: .webMercator()), scale: 2e7)
```

Comment:
There's no reason to go through the map view to get to the map that was created on line `24`. Could you please move the map creation into a `makeMap()` method like other samples have? You could make it a pure function by taking the feature layer as a parameter: `func makeMap(featureLayer: AGSFeatureLayer) -> AGSMap { ... }`.